target/arc/op_helper.c

```diff
@@ -187,10 +189,11 @@ uint64_t helper_llockd(CPUARCState *env, target_ulong addr)
     struct lpa_lf_entry *entry = &lpa_lfs[LPA_LFS_ENTRY_FOR_PA(haddr)];
     qemu_mutex_lock(&entry->mutex);
     env->arconnect.locked_mutex = &entry->mutex;
-    target_ulong read1 = cpu_ldl_data(env, addr);
-    target_ulong read2 = cpu_ldl_data(env, addr+4);
-    uint64_t ret = ((uint64_t ) read1) | (((uint64_t) read2) << 32);
     env->arconnect.lpa_lf = entry;
+
+    target_ulong read1 = cpu_ldl_data_ra(env, addr, GETPC());
```

Comment:
I think here we can use cpu_ldq_data_ra instead of 2 ldl ops.

Reply:
When fixing a bug that doesn't require restructuring the code I was told, and agree that, I shouldn't change more than absolutely necessary to fix the bug.
The little I change that goes beyond that is due to the patch style guide.
Wouldn't those changes you propose make more sense in a PR that restructures llock/scond variants, which we eventually will need for SMP? Or do you rather I do them in this PR.